File: packages/ruby/lib/opensms/resources/countries.rb
# frozen_string_literal: true

require_relative "base"

module Opensms
  module Resources
    # The +countries+ resource: the public country catalog. Accessed as
    # +client.countries+.
    class Countries < Base
      # GET /v1/countries -> Array<Country>.
      def list
        http_get("/v1/countries")
      end

      # GET /v1/countries/{iso2}/carriers -> Array<{ id:, name:, mcc_mnc:, prefixes: }>.
      def carriers(iso2)
        http_get(path("/v1/countries", id!(iso2, "iso2"), :carriers))
      end

      # GET /v1/countries/{iso2}/routes -> Array<Route>.
      def routes(iso2)
        http_get(path("/v1/countries", id!(iso2, "iso2"), :routes))
      end

      # GET /v1/countries/{iso2}/compliance -> CountryRules.
      def compliance(iso2)
        http_get(path("/v1/countries", id!(iso2, "iso2"), :compliance))
      end
    end
  end
end

File: packages/python/src/opensms/resources/countries.py
"""The ``countries`` resource."""

from __future__ import annotations

from typing import Any, List

from .._serialize import path_id
from ..models import Carrier, Country, CountryRules
from ._base import Resource


class Countries(Resource):
    """Public country catalog. Accessed as ``client.countries``."""

    def list(self) -> List[Country]:
        """Every supported country."""
        return self._t.request("GET", "/v1/countries")

    def carriers(self, iso2: str) -> List[Carrier]:
        """Carriers in a country."""
        return self._t.request("GET", f"/v1/countries/{path_id(iso2, 'iso2')}/carriers")

    def routes(self, iso2: str) -> List[Any]:
        """Routes available for a country."""
        return self._t.request("GET", f"/v1/countries/{path_id(iso2, 'iso2')}/routes")

    def compliance(self, iso2: str) -> CountryRules:
        """Compliance rules for a country."""
        return self._t.request("GET", f"/v1/countries/{path_id(iso2, 'iso2')}/compliance")
